kernel/data-pipeline/scenario/cdc/client/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/client/handler/RetryStreamingExceptionHandler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.data.pipeline.cdc.client.handler;
+
+import io.netty.channel.ChannelHandlerContext;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.data.pipeline.cdc.client.CDCClient;
+import org.apache.shardingsphere.data.pipeline.cdc.client.context.ClientConnectionContext;
+import org.apache.shardingsphere.data.pipeline.cdc.client.util.ServerErrorResult;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Retry streaming exception handler.
+ */
+@RequiredArgsConstructor
+@Slf4j
+public class RetryStreamingExceptionHandler implements ExceptionHandler {
+    
+    private final AtomicInteger retryTimes = new AtomicInteger(0);
+    
+    private CDCClient cdcClient;
+    
+    @Override
+    public void setCDCClient(final CDCClient cdcClient) {
+        this.cdcClient = cdcClient;
+    }
+    
+    @Override
+    public void handleServerException(final ChannelHandlerContext ctx, final ServerErrorResult result) {
+        // Server exception maybe not be resolved by retrying
+        log.error("Server error, code: {}, message: {}", result.getErrorCode(), result.getErrorMessage());
```

Comment:
Looks it's not defined clearly, users might need to customize it again, but could not just reuse RetryStreamingExceptionHandler.

Could we extract retry logic to utility class?

Or extract two handlers, one is exception handler, and another is server error handler?